Demo/Demo/ViewModels/PayPalViewModel.swift

```diff
@@ -0,0 +1,135 @@
+            }
+            accessToken = token
+            payPalClient = PayPalClient(config: CoreConfig(accessToken: token, environment: PaymentsCore.Environment.sandbox))
+            payPalClient?.delegate = self
+            state = .mainContent(title: "Access Token", content: accessToken, flowComplete: false)
+        }
+    }
+
+    func retry() {
+        payPalClient = nil
+        accessToken = ""
+        state = .initial
+    }
+
+    func checkoutWithOrder() {
+        startNativeCheckout {
+            let orderRequest = self.getOrderRequestUseCase.execute()
+            await self.payPalClient?.start { createOrderAction in
+                createOrderAction.create(order: orderRequest)
+            }
+        }
+    }
+
+    func checkoutWithOrderId() {
+        startNativeCheckout {
+            let orderId = try await self.getOrderIdUseCase.execute()
+            await self.payPalClient?.start { createOrderAction in
+                createOrderAction.set(orderId: orderId)
+            }
+        }
+    }
+
+    func checkoutWithBillingAgreement() {
+        startNativeCheckout {
+            let order = try await self.getBillingAgreementToken.execute()
+            await self.payPalClient?.start { createOrderAction in
+                createOrderAction.set(billingAgreementToken: order.id)
+            }
+        }
+    }
+
+    func checkoutBAWithoutPurchase() {
+        startNativeCheckout {
+            let baToken = try await self.getBATokenWithoutPurchaseUseCase.execute(accessToken: self.accessToken)
+            await self.payPalClient?.start { createOrderAction in
+                createOrderAction.set(billingAgreementToken: baToken)
+            }
+        }
+    }
+
+    func checkoutWithVault() {
+        startNativeCheckout {
+            guard let vaultSessionId = try await self.getApprovalSessionTokenUseCase.execute(accessToken: self.accessToken) else {
+                self.state = .mainContent(
+                    title: "Error",
+                    content: "Error in creating vault session!!",
+                    flowComplete: true
+                )
+
+                return
+            }
+            await self.payPalClient?.start { createOrderAction in
+                createOrderAction.set(vaultApprovalSessionID: vaultSessionId)
+            }
+        }
+    }
+
+    private func startNativeCheckout(withAction action: @escaping () async throws -> Void) {
+        state = .loading(content: "Initializing checkout")
+        Task {
+            do {
+                try await action()
+            } catch let error {
+                self.state = .mainContent(title: "Error", content: "\(error.localizedDescription)", flowComplete: true)
+            }
+        }
+    }
+
+    // MARK: - PayPalDelegate conformance
+
+    func paypalDidShippingAddressChange(
+        _ payPalClient: PayPalClient,
+        shippingChange: ShippingChange,
+        shippingChangeAction: ShippingChangeAction
+    ) {
+    }
```

Comment:
Is this a no-op or just missing implementation?

Reply:
Missing implementation for now. Going to add it on a separete PR

Reply:
Let's add a TODO so this doesn't get forgotten.